### ISSUE TITLE
JSON.stringify(undefined) is not a string, but Typescript only defines the return type of string

### ISSUE THREAD

**Issue by huboshu0819** [microsoft/TypeScript#40296]
**TypeScript Version:**  3.7.5

**Search Terms:**

**Code**

```ts
JSON.stringify(undefined).replace(/aa/ig, '');
```

**Expected behavior: typescript should prompt a type error

**Actual behavior: No error message

**Playground Link:** 

**Related Issues:**

**Comment by huboshu0819** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

**Comment by RyanCavanaugh** [comment]
Duplicate #18879
